cli/src/main.rs

```diff
@@ -0,0 +1,221 @@
+        self.db.put(key.as_bytes(), value.as_bytes()).await;
+        Ok(())
+    }
+
+    async fn get(&self, key: &str) -> Result<Option<String>, Box<dyn std::error::Error>> {
+        if let Some(bytes) = self.db.get(key.as_bytes()).await? {
+            let value = String::from_utf8(bytes.to_vec())?;
+            Ok(Some(value))
+        } else {
+            Ok(None)
+        }
+    }
+
+    async fn delete(&self, key: &str) -> Result<(), Box<dyn std::error::Error>> {
+        self.db.delete(key.as_bytes()).await;
+        Ok(())
+    }
+
+    async fn close(&self) -> Result<(), Box<dyn std::error::Error>> {
+        self.db.close().await?;
+        Ok(())
+    }
+
+    fn help() {
+        println!("Available commands:");
+        println!("  put <key> <value>    - Insert a key-value pair into SlateDB");
+        println!("  get <key>            - Retrieve the value associated with a key");
+        println!("  delete <key>         - Delete a key-value pair from SlateDB");
+        println!("  help                 - Show this help message");
+        println!("  exit | quit          - Exit");
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), ReadlineError> {
+    let args = Args::parse();
+
+    let db_cli = match SlateDbREPL::new(&args.storage_path).await {
+        Ok(cli) => cli,
+        Err(e) => {
+            eprintln!("Failed to initialize the database: {}", e);
+            std::process::exit(1);
+        }
+    };
+
+    // Initialize the REPL
+    let mut rl = DefaultEditor::new()?;
+
+    println!("Welcome to the SlateDB REPL!");
+    println!("WARNING: This is currently intended for development use.");
+    println!(
+        r#"
+  ____  _       _       ____  ____
+ / ___|| | __ _| |_ ___|  _ \| __ )
+ \___ \| |/ _` | __/ _ \ | | |  _ \
+  ___) | | (_| | ||  __/ |_| | |_) |
+ |____/|_|\__,_|\__\___|____/|____/
+"#
+    );
+    println!("Type 'help' to see available commands.\n");
+
+    loop {
+        // Prompt the user for input
+        let readline = rl.readline(">> ");
+        match readline {
+            Ok(line) => {
+                let trimmed = line.trim();
+                if trimmed.is_empty() {
+                    continue;
+                }
+                match rl.add_history_entry(trimmed) {
+                    Ok(_) => {}
+                    Err(e) => {
+                        eprintln!("Could not save line to history: {}", e)
+                    }
+                }
+
+                // Parse the input using shell-words to handle quotes
+                let parts_result = split(trimmed);
+                let parts = match parts_result {
+                    Ok(p) => p,
+                    Err(e) => {
+                        eprintln!("Error parsing input: {}", e);
+                        continue;
+                    }
+                };
+
+                if parts.is_empty() {
+                    continue;
+                }
```

Comment:
can this snippet be moved up and the continue combined with the `trimmed.is_empty()`? and then we can check for history entry and we can remove the `parts_result` intermediate variable as its used only once

like this

```rust
let trimmed = line.trim();
let parts = match split(trimmed) {...}
if trimmed.is_empty() || parts.is_empty() {
       continue;
}

// then rl.add_history_entry(trimmed) .......
```

seems a bit cleaner to me if it doesn't break any logic